appcues/src/main/java/com/appcues/AppcuesProviderService.kt

```diff
@@ -0,0 +1,52 @@
+package com.appcues
+
+import android.app.Activity
+import android.app.Application
+import android.content.Context
+import com.appcues.monitor.AppcuesActivityMonitor
+
+/**
+ * Allows programmatic access to start and stop services normally controlled
+ * through an androidx.startup.InitializationProvider. In most typical use cases,
+ * this access is not expected to be necessary, and these services will start
+ * automatically.
+ */
+public object AppcuesProviderService {
+
+    /**
+     * Start Appcues services.
+     *
+     * It is not normally expected to call this function, as services are started
+     * automatically using an androidx.startup.InitializationProvider. However, if necessary
+     * to have programmatic control to stop and start services, this function can be used.
+     *
+     * The [context] parameter should be the currently active Activity, if possible, to ensure
+     * that the Appcues SDK has this Activity available to present content.
+     *
+     * @param context The Android Context used by the host application.
+     */
+    public fun start(context: Context) {
+        // stop any existing services first before starting new
+        stop(context)
+
+        val application = context.applicationContext as Application
```

Comment:
First of all, thank you for accepting our request. In our experience we have had crashes due to application context null, for example when initializing a library in `onAttachBaseContext`. personally I would suggest doing this check and logging the exception, what do you think?

Reply:
Good call - added a check for this in the commit [here](https://github.com/appcues/appcues-android-sdk/pull/518#:~:text=Resolve%20conversation-,%F0%9F%91%8C%20Add%20check%20for%20null%20application%20context,-f71d276). Would that work ok?

Reply:
perfect 👌

Reply:
it's perfect as in pull request, for us the important thing was only to avoid the crash and have a log entry :)